src/common/collection.c

```diff
@@ -1580,9 +1580,22 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
       query = dt_util_dstrcat(query, ")");
       break;
     case DT_COLLECTION_PROP_TAG: // tag
-      query = dt_util_dstrcat(query, "(id IN (SELECT imgid FROM main.tagged_images AS a JOIN "
+    
+      /* shift-click adds an asterix * to include items in and under this hierarchy without using a wildcard % which also would include similar named items */
+      if (!strcmp(strrchr(escaped_text, '\0') - 1, "*"))
```

Comment:
if escaped_text is an empty string, -1 will point to the char before the actual string which is wrong and could not even be allocated. Use strchr(... '*') and check that it is the last char strchr == strlen or something like that.

Reply:
@TurboGit : thanks a lot for the hint, I just pushed a fix